lib/android/app/src/main/java/com/reactnativenavigation/anim/TopBarAnimator.java

```diff
@@ -0,0 +1,77 @@
+package com.reactnativenavigation.anim;
+
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.animation.ObjectAnimator;
+import android.animation.TimeInterpolator;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.AccelerateInterpolator;
+import android.view.animation.DecelerateInterpolator;
+
+import com.reactnativenavigation.views.TopBar;
+
+public class TopBarAnimator {
+
+    private static final int DURATION_TOPBAR = 300;
+    private DecelerateInterpolator decelerateInterpolator;
+    private AccelerateInterpolator accelerateInterpolator;
+
+    public TopBarAnimator() {
+        decelerateInterpolator = new DecelerateInterpolator();
+        accelerateInterpolator = new AccelerateInterpolator();
+    }
+
+    public void animateShowTopBar(final TopBar topBar, final View contentView) {
```

Comment:
Both TopBar and contentView should be params of `TopBarAnimator`. This way you're end up with a class which exposes public commands that only accept variables that describe the command, like start translation and duration. This makes the code much easier to understand.